persistent/Database/Persist/Quasi/Internal/ModelParser.hs

```diff
@@ -1,11 +1,15 @@
 {-# LANGUAGE DeriveLift #-}
+{-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE TupleSections #-}
 
 module Database.Persist.Quasi.Internal.ModelParser
-    ( SourceLoc (..)
+    ( ParserSettings (..)
+    , defaultParserSettings
```

Comment:
We'll need to expose a safe interface for `ParserSettings` from `Database.Persist.Quasi`

Suggest creating new module `Database.Persist.Quasi.ParserSettings` which defines the interface + `Database.Persist.Quasi.ParserSettings.Internal` which exports everything